File: src/priv/bajdcc/LL1/syntax/prediction/PredictionInstruction.java
package priv.bajdcc.LL1.syntax.prediction;

import priv.bajdcc.LL1.syntax.token.PredictType;

/**
 * 【LL1预测分析】预测分析指令结构
 *
 * @author bajdcc
 */
public class PredictionInstruction {

	/**
	 * 指令类型
	 */
	public PredictType type;

	/**
	 * 指令数据
	 */
	public int inst;

	public PredictionInstruction(PredictType type, int inst) {
		this.type = type;
		this.inst = inst;
	}

	@Override
	public boolean equals(Object obj) {
		if (obj instanceof PredictionInstruction) {
			PredictionInstruction pi = (PredictionInstruction) obj;
			return type == pi.type && inst == pi.inst;
		}
		return false;
	}

	@Override
	public String toString() {
		return type.getName() + ": " + inst;
	}
}

File: src/main/kotlin/com/bajdcc/LL1/syntax/prediction/PredictionInstruction.kt
package com.bajdcc.LL1.syntax.prediction

import com.bajdcc.LL1.syntax.token.PredictType

/**
 * 【LL1预测分析】预测分析指令结构
 *
 * @author bajdcc
 */
class PredictionInstruction(var type: PredictType,
                            var inst: Int) {

    override fun equals(other: Any?): Boolean {
        if (other is PredictionInstruction) {
            val pi = other as PredictionInstruction?
            return type == pi!!.type && inst == pi.inst
        }
        return false
    }

    override fun toString(): String {
        return type.desc + ": " + inst
    }

    override fun hashCode(): Int {
        var result = type.hashCode()
        result = 31 * result + inst
        return result
    }
}
